nucypher/characters/lawful.py

```diff
@@ -730,28 +718,83 @@ def work_orders_for_capsules(self,
                 # TODO: Presently, the order here is haphazard .  Do we want to do the complete or incomplete specifically first? NRN
                 break
 
-        if incomplete_work_orders == OrderedDict():
-            self.log.warn("No new WorkOrders created. Try calling this with different parameters.")  # TODO: Clearer instructions.  NRN
+        if not incomplete_work_orders:
+            self.log.warn(
+                "No new WorkOrders created.  Try calling this with different parameters.")  # TODO: Clearer instructions.  NRN
 
         return incomplete_work_orders, complete_work_orders
 
-    def get_reencrypted_cfrags(self, work_order, retain_cfrags=False):
-        if work_order.completed:
-            raise TypeError("This WorkOrder is already complete; "
-                            "if you want Ursula to perform additional service, make a new WorkOrder.")
-
-        cfrags_and_signatures = self.network_middleware.reencrypt(work_order)
-        cfrags = work_order.complete(cfrags_and_signatures)
-        self._completed_work_orders.save_work_order(work_order, as_replete=retain_cfrags)
-
-        return cfrags
-
     def join_policy(self, label, alice_verifying_key, node_list=None, block=False):
         if node_list:
             self._node_ids_to_learn_about_immediately.update(node_list)
         treasure_map = self.get_treasure_map(alice_verifying_key, label)
         self.follow_treasure_map(treasure_map=treasure_map, block=block)
 
+    def _filter_work_orders_and_capsules(self, work_orders, capsules, m):
```

Comment:
Should we provide a type hint for our return values here `-> Tuple [List, Set]`?

Reply:
Done. Added more typing here and in `_reencrypt()` as well.